src/stores/events/EventsDataStore.ts

```diff
@@ -317,21 +343,24 @@ export default class EventsDataStore {
 		}
 	};
 
+	@action
 	private parentNodesUpdater = () => {
 		const parentNodes = this.loadedParentNodes;
 		if (!parentNodes.length) return;
 
 		const parentChildrenMapUpdate: Map<string, string[]> = new Map();
-		const parentLoadingStatusUpdate: Map<string, boolean> = new Map();
 		const rootNodes: string[] = [];
 
 		parentNodes.forEach(parentNodePath => {
 			for (let i = 0; i < parentNodePath.length; i++) {
 				const event = parentNodePath[i];
 
-				const { isUnknown, parentId, eventId } = event;
+				const { parentId: realParentId, eventId } = event;
+				const cachedParent = realParentId ? this.eventsCache.get(realParentId) : null;
+				// We need this to change the parent to an unknown root if needed
+				const parentId = cachedParent?.eventId ?? event.parentId;
```

Comment:
same problem, i don't understand why do we need to get parent event from cache, if in the end it will be `event.parentId`